tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -113,6 +114,43 @@ func newSession(credentialPath, credentialAccountID, region string) (*awssession
 	})
 }
 
+// retrievePrivateKey retrieves private key in format required for generating
+// public key using the privateKeyPath
+func (a *AwsProvider) retrievePrivateKey() (*rsa.PrivateKey, error) {
+	privateKeyBytes, err := ioutil.ReadFile(a.privateKeyPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find private key from path: %v", a.privateKeyPath)
+	}
+
+	privateKeyBlock, _ := pem.Decode(privateKeyBytes)
+
+	var parsedKey interface{}
+	if parsedKey, err = x509.ParsePKCS1PrivateKey(privateKeyBlock.Bytes); err != nil {
+		return nil, fmt.Errorf("failed to parse private key with: %v", err)
+	}
+
+	privateKey, ok := parsedKey.(*rsa.PrivateKey)
+	if !ok {
+		return nil, fmt.Errorf("failed to convert private key to rsa key type: %v", err)
+	}
+	return privateKey, nil
+}
+
+// generatePublicKey generates a public key and returns it in the ssh-rsa..
+// string format using the retrieved private key
+func (a *AwsProvider) generatePublicKey() (string, error) {
+	privateKey, err := a.retrievePrivateKey()
+	if err != nil {
+		return "", fmt.Errorf("failed to retrieve private key in rsa format: %v", err)
+	}
+	publicRsaKey, err := ssh.NewPublicKey(&privateKey.PublicKey)
```

Comment:
`publicRsaKey` -> `publicRSAKey`

Reply:
I would suggest `publicKey`. The key needn't use RSA.